Amazon SQS Example

john-jac opened the pull request:
In this PR I've added a doc entry and example for using the SQS Publish Operator.

The example included creating a new SQS queue, publishing to that queue, reading the queue, and deleting the queue.

The documentation refers to the example.

john-jac committed:
Amazon SQS Example

Added a doc entry and example for using the SQS Publish Operator

mik-laj commented on airflow/providers/amazon/aws/example_dags/example_sqs.py:
```diff
@@ -0,0 +1,83 @@
+AWS_CONN_ID = 'aws_default'
+
+
+def create_queue_fn(**kwargs):
```
Do you consider using [Task Flow API](https://airflow.apache.org/docs/apache-airflow/stable/tutorial_taskflow_api.html)?

john-jac replied:
I did...and I was torn because the other examples don't use it, but this one would be a good candidate.  Would you recommend I switch it to Task Flow?

uranusjr replied:
The Task Flow API is relatively new, a lot of the examples were written even before its introduction, and it isn't worth the effort to convert them all. I'd definitely recommend using it for new examples if it fits the need.

john-jac replied:
OK sounds good.  Let me go ahead and switch to Task Flow and update the PR

mik-laj replied:
Update for example DAGs is WIP. https://github.com/apache/airflow/pull/16837
Cc: @josh-fell

josh-fell replied:
@john-jac There weren't a lot of provider example DAGs that warranted an update to use the TaskFlow API but #18278 converted the applicable ones.

josh-fell replied:
There is also an ongoing effort to transition away from using `days_ago(n)` for `start_date` to a static value as well cleaning up `default_args` usage so they are pertinent for the example. #18597 is an example of these updates for reference.

john-jac replied:
Thanks folks for all of the help!  I've modified the example to use Task Flow API and I added a missing doc reference in the operator source code.

john-jac committed:
Changed Amazon SQS Example to Task Flow API and added ref in sqs.py

The example now uses the TaskFlow API to pass results from the Python functions, and per the build error is now referred to in the operator source code.

potiuk reviewed (approved):
Nice example :)
